llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1846,7 +1846,7 @@ static void removeBranchOnCondTrue(VPlan &Plan) {
   using namespace llvm::VPlanPatternMatch;
   for (VPBasicBlock *VPBB : VPBlockUtils::blocksOnly<VPBasicBlock>(
            vp_depth_first_shallow(Plan.getEntry()))) {
-    if (VPBB->getNumSuccessors() != 2 ||
+    if (VPBB->getNumSuccessors() != 2 || isa<VPIRBasicBlock>(VPBB) ||
```

Comment:
Excluding VPIRBB's only because of the temporary terminator-less entry block with two successors? If so, suffice to exclude `VPBB == Plan.getEntry()`? But from an earlier comment: "we still need to allow terminator-less VPIRBBs for various parts of the skeleton."

Reply:
Updated, thanks! We don't run removeBranchOnCond after introducing the other, terminator-less VPIRBBs.